app/routing/DeparturesRouter.scala

```diff
@@ -51,19 +51,28 @@ class DeparturesRouter @Inject() (
     case _                                                    => v1Departures.submitDeclaration()
   }
 
-  def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] = route {
-    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) =>
-      runIfBound[V2DepartureId](
-        "departureId",
-        departureId,
-        boundDepartureId => runIfBound[V2MessageId]("messageId", messageId, v2Departures.getMessage(boundDepartureId, _))
-      )
-    case _ =>
-      runIfBound[V1DepartureId](
-        "departureId",
-        departureId,
-        boundDepartureId => runIfBound[V1MessageId]("messageId", messageId, v1DepartureMessages.getDepartureMessage(boundDepartureId, _))
-      )
+  def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] = {
+    def routeToVersion2(convertBodyToJson: Boolean) = runIfBound[V2DepartureId](
+      "departureId",
+      departureId,
+      boundDepartureId =>
+        runIfBound[V2MessageId](
+          "messageId",
+          messageId,
+          v2Departures.getMessage(boundDepartureId, _, convertBodyToJson)
+        )
+    )
+
+    route {
+      case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE)          => routeToVersion2(false)
+      case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML) => routeToVersion2(true)
+      case _ =>
+        runIfBound[V1DepartureId](
+          "departureId",
+          departureId,
+          boundDepartureId => runIfBound[V1MessageId]("messageId", messageId, v1DepartureMessages.getDepartureMessage(boundDepartureId, _))
+        )
+    }
```

Comment:
I think the router should only be concerned about routing to the correct version, and not more than that. The reason for this is that when we retire version 1 and delete all the v1 code, this router will likely go with it, and as it currently stands, you've got v2 specific logic in here.

I think the better solution is to pattern match on the regex `^application/vnd\.hrmc\.2\.0\+.+$` and let the v2 controller, rather than the router, dictate the return type -- that way when this router gets removed, the logic is still there.

Reply:
Also, additionally, if I set the accept header to be `application/vnd.hmrc.2.0+xml`, which may become a valid option in the future, this will redirect to the v1 route, though I think that's a problem with all routing we have and we might need to look into that...

Reply:
@drnaylor-oc yep agreed, good point